src/core/master/src/main/java/org/drftpd/master/network/BaseFtpConnection.java

```diff
@@ -332,19 +332,18 @@ public void run() {
             _thread.setName("FtpConn thread " + _thread.getId() + " from " + getClientAddress().getHostAddress());
         }
 
-        _pool = new ThreadPoolExecutor(1, Integer.MAX_VALUE, 60L, TimeUnit.SECONDS, new SynchronousQueue<>(), new CommandThreadFactory(_thread.getName()));
+        _pool = new ThreadPoolExecutor(1, 1, 60L, TimeUnit.SECONDS, new LinkedBlockingQueue<>(), new CommandThreadFactory(_thread.getName()));
```

Comment:
Why do you need a pool executor anymore?
If you want a sequential execution of every commands, you should just remove it.

Reply:
The FTP protocol demands that an "ABOR" command can be given at any time and needs to be handled immediately.
The executor is to ensure the I/O thread is ready to handle whatever.

The logic of the code was to ensure single threaded ness by dropping/discarding messages, which is a clear bad idea at this stage (it seems).

Currently the easiest solution to ensure commands are handled in the order they came in (excluding ABOR) is to use a FIFO queue and a single threaded executor.

I agree it is less than ideal, but at least improves the way drftpd handles the control connection.

Reply:
Ok thanks for the explanation.
Looks like a good trade off with the current code base, good job.

Reply:
Ok thanks for the explanation.
Looks like a good trade off with the current code base, good job.